### DISCUSSION TITLE
Error: (sqlite3.DataError) string or blob too big and sqlite3.OperationalError after updating from 0.3.28 to 0.3.35

### DISCUSSION THREAD

**Discussion by YetheSamartaka** [open-webui/open-webui#6755] [Issues]
# Bug Report

## Installation Method

[Docker]

## Environment

- **Open WebUI Version:** [v0.3.35]

- **Operating System:** [Ubuntu 22.04]
- **Browser (if applicable):** [Firefox, Chrome, Brave, Edge]

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

There is no such error and after logging in, I do not have a blank screen and I can continue using WebUI as before the update

## Actual Behavior:

Blank screen after logging in.

## Description

**Bug Summary:**
After updating from from 0.3.28 to 0.3.35, the WebUI no longer works after user is logged in. The Login page is alright, but after logging in, there is a blank screen. I found Error: (sqlite3.DataError) string or blob too big and sqlite3.OperationalError in logs (mentioned below) as errors.

## Reproduction Details

**Steps to Reproduce:**
Update from 0.3.28 to 0.3.35 where you were using the WebUI before quite a lot so you have multiple users with different chats already stored.

## Logs and Screenshots

**Docker Container Logs:**
At the beginning of the log:

Running migrations
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade 6a39f3d8e55c -> 242a2047eae0, Update chat table
INFO  [open_webui.env] 'JWT_EXPIRES_IN' loaded from the latest database entry
INFO  [open_webui.env] 'OLLAMA_BASE_URLS' loaded from the latest database entry
INFO  [open_webui.env] 'OPENAI_API_KEYS' loaded from the latest database entry
INFO  [open_webui.env] 'OPENAI_API_BASE_URLS' loaded from the latest database entry
INFO  [open_webui.env] 'ENABLE_SIGNUP' loaded from the latest database entry
INFO  [open_webui.env] 'DEFAULT_MODELS' loaded from the latest database entry
INFO  [open_webui.env] 'DEFAULT_PROMPT_SUGGESTIONS' loaded from the latest database entry
INFO  [open_webui.env] 'DEFAULT_USER_ROLE' loaded from the latest database entry
INFO  [open_webui.env] 'USER_PERMISSIONS' loaded from the latest database entry
INFO  [open_webui.env] 'ENABLE_MODEL_FILTER' loaded from the latest database entry
INFO  [open_webui.env] 'MODEL_FILTER_LIST' loaded from the latest database entry
INFO  [open_webui.env] 'WEBHOOK_URL' loaded from the latest database entry
INFO  [open_webui.env] 'ENABLE_COMMUNITY_SHARING' loaded from the latest database entry
INFO  [open_webui.env] 'WEBUI_BANNERS' loaded from the latest database entry
INFO  [open_webui.env] 'SHOW_ADMIN_DETAILS' loaded from the latest database entry
INFO  [open_webui.env] 'TASK_MODEL' loaded from the latest database entry
INFO  [open_webui.env] 'TASK_MODEL_EXTERNAL' loaded from the latest database entry
INFO  [open_webui.env] 'TITLE_GENERATION_PROMPT_TEMPLATE' loaded from the latest database entry
INFO  [open_webui.env] 'ENABLE_SEARCH_QUERY' loaded from the latest database entry
INFO  [open_webui.env] 'SEARCH_QUERY_GENERATION_PROMPT_TEMPLATE' loaded from the latest database entry
INFO  [open_webui.env] 'TOOLS_FUNCTION_CALLING_PROMPT_TEMPLATE' loaded from the latest database entry
INFO  [open_webui.env] 'CONTENT_EXTRACTION_ENGINE' loaded from the latest database entry
INFO  [open_webui.env] 'TIKA_SERVER_URL' loaded from the latest database entry
INFO  [open_webui.env] 'RAG_TOP_K' loaded from the latest database entry
INFO  [open_webui.env] 'RAG_RELEVANCE_THRESHOLD' loaded from the latest database entry
INFO  [open_webui.env] 'ENABLE_RAG_HYBRID_SEARCH' loaded from the latest database entry
INFO  [open_webui.env] 'RAG_EMBEDDING_ENGINE' loaded from the latest database entry
INFO  [open_webui.env] 'PDF_EXTRACT_IMAGES' loaded from the latest database entry
INFO  [open_webui.env] 'RAG_EMBEDDING_MODEL' loaded from the latest database entry
INFO  [open_webui.env] Embedding model set: mxbai-embed-large:latest
INFO  [open_webui.env] 'RAG_RERANKING_MODEL' loaded from the latest database entry
INFO  [open_webui.env] Reranking model set: mixedbread-ai/mxbai-rerank-large-v1
INFO  [open_webui.env] 'CHUNK_SIZE' loaded from the latest database entry
INFO  [open_webui.env] 'CHUNK_OVERLAP' loaded from the latest database entry
INFO  [open_webui.env] 'RAG_TEMPLATE' loaded from the latest database entry
INFO  [open_webui.env] 'RAG_OPENAI_API_BASE_URL' loaded from the latest database entry
INFO  [open_webui.env] 'RAG_OPENAI_API_KEY' loaded from the latest database entry
INFO  [open_webui.env] 'YOUTUBE_LOADER_LANGUAGE' loaded from the latest database entry
INFO  [open_webui.env] 'ENABLE_RAG_WEB_SEARCH' loaded from the latest database entry
INFO  [open_webui.env] 'RAG_WEB_SEARCH_ENGINE' loaded from the latest database entry
INFO  [open_webui.env] 'SEARXNG_QUERY_URL' loaded from the latest database entry
INFO  [open_webui.env] 'GOOGLE_PSE_API_KEY' loaded from the latest database entry
INFO  [open_webui.env] 'GOOGLE_PSE_ENGINE_ID' loaded from the latest database entry
INFO  [open_webui.env] 'BRAVE_SEARCH_API_KEY' loaded from the latest database entry
INFO  [open_webui.env] 'SERPSTACK_API_KEY' loaded from the latest database entry
INFO  [open_webui.env] 'SERPSTACK_HTTPS' loaded from the latest database entry
INFO  [open_webui.env] 'SERPER_API_KEY' loaded from the latest database entry
INFO  [open_webui.env] 'SERPLY_API_KEY' loaded from the latest database entry
INFO  [open_webui.env] 'TAVILY_API_KEY' loaded from the latest database entry
INFO  [open_webui.env] 'RAG_WEB_SEARCH_RESULT_COUNT' loaded from the latest database entry
INFO  [open_webui.env] 'RAG_WEB_SEARCH_CONCURRENT_REQUESTS' loaded from the latest database entry
INFO  [open_webui.env] 'IMAGE_GENERATION_ENGINE' loaded from the latest database entry
INFO  [open_webui.env] 'ENABLE_IMAGE_GENERATION' loaded from the latest database entry
INFO  [open_webui.env] 'IMAGES_OPENAI_API_BASE_URL' loaded from the latest database entry
INFO  [open_webui.env] 'IMAGES_OPENAI_API_KEY' loaded from the latest database entry
INFO  [open_webui.env] 'IMAGE_SIZE' loaded from the latest database entry
INFO  [open_webui.env] 'IMAGE_STEPS' loaded from the latest database entry
INFO  [open_webui.env] 'IMAGE_GENERATION_MODEL' loaded from the latest database entry
INFO  [open_webui.env] 'AUDIO_STT_OPENAI_API_BASE_URL' loaded from the latest database entry
INFO  [open_webui.env] 'AUDIO_STT_OPENAI_API_KEY' loaded from the latest database entry
INFO  [open_webui.env] 'AUDIO_STT_ENGINE' loaded from the latest database entry
INFO  [open_webui.env] 'AUDIO_STT_MODEL' loaded from the latest database entry
INFO  [open_webui.env] 'AUDIO_TTS_OPENAI_API_BASE_URL' loaded from the latest database entry
INFO  [open_webui.env] 'AUDIO_TTS_OPENAI_API_KEY' loaded from the latest database entry
INFO  [open_webui.env] 'AUDIO_TTS_API_KEY' loaded from the latest database entry
INFO  [open_webui.env] 'AUDIO_TTS_ENGINE' loaded from the latest database entry
INFO  [open_webui.env] 'AUDIO_TTS_MODEL' loaded from the latest database entry
INFO  [open_webui.env] 'AUDIO_TTS_VOICE' loaded from the latest database entry
WARNI [chromadb.db.impl.sqlite] ⚠️ It looks like you upgraded from a version below 0.6 and could benefit from vacuuming your database. Run chromadb utils vacuum --help for more information.

Error: (sqlite3.DataError) string or blob too big
[SQL: UPDATE chat SET chat=? WHERE chat.id = ?]
[parameters: ('{"id": "", "title": "\\ud83c\\udfdd\\ufe0f Baltic Coastline Design", "models": ["gpt-4o"], "params": {}, "history": {"messages": {"784437c9-187b-4b45 ... (520115752 characters truncated) ... "prompt_tokens_details": {"cached_tokens": 1408}, "completion_tokens_details": {"reasoning_tokens": 0}, "openai": true}, "done": true}], "files": []}', '482fe9ba-0b80-4f9b-ad8f-2fe3c0497bad')]
(Background on this error at: https://sqlalche.me/e/20/9h9h)

And then after some time, there is this:

ERROR [open_webui.apps.webui.routers.chats] (sqlite3.OperationalError) no such column: tag.meta
[SQL: SELECT tag.id AS tag_id, tag.name AS tag_name, tag.user_id AS tag_user_id, tag.meta AS tag_meta
FROM tag
WHERE tag.user_id = ?]
[parameters: ('a928f667-6355-4692-8988-708a174e01ec',)]
(Background on this error at: https://sqlalche.me/e/20/e3q8)
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1967, in _exec_single_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 941, in do_execute
    cursor.execute(statement, parameters)
sqlite3.OperationalError: no such column: tag.meta

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/backend/open_webui/apps/webui/routers/chats.py", line 233, in get_all_user_tags
    tags = Tags.get_tags_by_user_id(user.id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backend/open_webui/apps/webui/models/tags.py", line 82, in get_tags_by_user_id
    for tag in (db.query(Tag).filter_by(user_id=user_id).all())
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/query.py", line 2673, in all
    return self._iter().all()  # type: ignore
           ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/query.py", line 2827, in _iter
    result: Union[ScalarResult[_T], Result[_T]] = self.session.execute(
                                                  ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 2362, in execute
    return self._execute_internal(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 2247, in _execute_internal
    result: Result[Any] = compile_state_cls.orm_execute_statement(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/context.py", line 293, in orm_execute_statement
    result = conn.execute(
             ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1418, in execute
    return meth(
           ^^^^^
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/sql/elements.py", line 515, in _execute_on_connection
    return connection._execute_clauseelement(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1640, in _execute_clauseelement
    ret = self._execute_context(
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1846, in _execute_context
    return self._exec_single_context(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1986, in _exec_single_context
    self._handle_dbapi_exception(
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 2355, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1967, in _exec_single_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 941, in do_execute
    cursor.execute(statement, parameters)

## Additional Information

When I was trying to search for it, I found this closed report, so I am not the only one with this discussion:
https://github.com/open-webui/open-webui/discussions/6394

**Comment by tjbck** [comment]
Help wanted here, keep us updated with your troubleshooting journey!

**Reply by YetheSamartaka** [reply]
Well, I'm currently contemplating between increasing limits for SQLite, but when looking at the scale for which the Open WebUI is going to be used in my application, I'm more leaning towards the migration to PostgreSQL and connecting the WebUI to it. It would be wise to implement some kind of error message to be displayed to users when such issue with database occurrs instead of a blank screen. I hope that the migration will solve the first issue and possibly even the second one which might be caused by not completing the migration in the first place. And even if the second issue would not be resolved by migration, I can just add that "meta" column to the "tag" manually.

**Comment by YetheSamartaka** [comment]
@tjbck Unfortunately after more than a month of struggle, I was not successful with my team to properly migrate the database or resolve this issue. Migration fails on the premise that the migrated data is too big. The whole .db has 1,4 GB. We tried to the latest 0.4.7 version of the WebUI, still the same issue. 

We tried migrating SQLite db to Postgres via pgloader, sqlalchemy and also some manual ways, but unsuccessfully. My main domain is far far away from databases. Shouldn't the WebUI migration scripts themselves be equipped to handle these "large amounts" of data when upgrading to the new version? We have two environments, one production for the broad usage, other for the dev. Since the dev one is significantly less used, upgrading to newer versions is without issues but the production one is currently stuck.
